src/librustc/hir/lowering.rs

```diff
@@ -3143,17 +3139,27 @@ impl<'a> LoweringContext<'a> {
                     });
                 }
 
-                // Privatize the degenerate import base, used only to check
-                // the stability of `use a::{};`, to avoid it showing up as
-                // a re-export by accident when `pub`, e.g. in documentation.
                 let def = self.expect_full_def_from_use(id).next().unwrap_or(Def::Err);
                 let path = P(self.lower_path_extra(def, &prefix, ParamMode::Explicit, None));
-                *vis = respan(prefix.span.shrink_to_lo(), hir::VisibilityKind::Inherited);
```

Comment:
IIRC, you still need to reset at least `pub` (`hir::VisibilityKind::Public`) into something more private, otherwise you'll get *a lot* of noise from rustdoc (the stem is emitted for all braced imports).

This way the hack for `UnreachablePub` can be avoided as well.

This should be ok since `hir::VisibilityKind::Public` doesn't have a path with node IDs inside it.

Reply:
Everything except for `VisibilityKind::Restricted` can be reset to `VisibilityKind::Inherited` basically.